meta-lxatac-software/recipes-devtools/gitlab-runner/gitlab-runner.inc

```diff
@@ -0,0 +1,29 @@
+SUMMARY = "GitLab Runner"
+DESCRIPTION = "The Runner for GitLab Pipelines"
+
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+
+SRC_URI:append = " file://gitlab-runner.service "
+
+S = "${WORKDIR}/git"
+
+GO_IMPORT = "gitlab.com/gitlab-org/gitlab-runner"
+
+RDEPENDS:gitlab-runner-dev = "bash"
+
+inherit go
+inherit go-mod
+inherit systemd
+
+SYSTEMD_SERVICE:${PN} = "gitlab-runner.service"
+
+do_compile() {
+    export TMPDIR="${GOTMPDIR}"
+    ${GO} install ${GOBUILDFLAGS}
+}
```

Comment:
This needs a comment why the default from the go classes is not sufficient.

Reply:
Oh yeah, I kind of forgot that I had to add that. I've updated the commit and moved the workaround a bit closer to the actual culprit:

https://github.com/linux-automation/meta-lxatac/blob/755c3c23c9399ad2e591bf413ab30666caa1a36c/meta-lxatac-software/recipes-devtools/gitlab-runner/gitlab-runner.inc#L21-L30

I don't really have a good answer as to _why_ the compile stage fails, because I don't know to much about Go and how to package Go projects.

Here is an excerpt from the do_compile logs:

```
DEBUG: Executing shell function do_compile
HASH[moduleIndex]
HASH[moduleIndex]: "go1.19.7"
HASH[moduleIndex]: "modroot /ptx/work/WORK_SANAH/lgo/Projects/meta-lxatac/build/tmp/work/cortexa7t2hf-neon-vfpv4-oe-linux-gnueabi/github-act-runner/0.6.5-r0/build/src/github.com/ChristopherHX/github-act-runner\n"
… 40.000 similar lines …
| HASH[link gitlab.com/gitlab-org/gitlab-runner]: "packagefile github.com/mailru/easyjson/buffer=sinqWKS3te11-zQA-HqR\n"
| HASH[link gitlab.com/gitlab-org/gitlab-runner]: 51391287bfd80a3cf807d8f80fa8d621e6d0b02c9a095de9da62b2afc6b68609
| HASH subkey 4eba7844681da8d69748f63e968ed43b568face86d54d7a118abda3a51623ff5 "link-stdout" = c4e9afa14a7d97bef211d92e76f7d06c0e608a848d8d54e01528885ad669bb53
| HASH subkey 6b98acb0b9b3698479e89ec43fe407ed39e0972fe600d46b6456c6cedc82226f "link-stdout" = 6b84cae771da7444fd35ad8e010da0dc707687a1b07a4bce58c1530f04f37617
| HASH subkey 51806134918812b94be74eb1507758c70ec7d2c4abaa90cd3a121a6bf2a7fe98 "link-stdout" = f4db7eb852d9f0b26c3cc3a97f1b834a6a7a3bed0c7b86b0e71a8b02548c1125
| HASH subkey 5a95087a85e736d91c9245885f3e51a83c14fdbae73d12e98ecc6356b69e8476 "link-stdout" = 64c050825db8d5d4a74d35aabc117f02c62399fe873178b1633026f6bbe9212e
| HASH subkey 80cda6fa8385d2022e4a2cca1cafab81bfd235b4b6a0244b41842eec5e938072 "link-stdout" = 43db68b16b5720db3c90e10490d3ee19a204e1962ca880de5d660c128688f553
| HASH subkey 39a95685e237b046cf7bb71e6dc2b8118ee450b7fabde6ebc0d966efcf2a7b9e "link-stdout" = effaa42e9e940d9b2b22326f658cfcd1db9ff2e9fec0086f4df1e1a7d551d0ea
| HASH subkey 7a01f21523fee0823f613cb5be41074b3f0afa10ae82bd9a124c8cb03030d487 "link-stdout" = 74150482e2cd43588d973ede4ead023f349525e2d1f1f57b195206e9c816aa66
| WARNING: exit code 2 from a shell command.
ERROR: Task (/ptx/work/WORK_SANAH/lgo/Projects/meta-lxatac/poky/../meta-lxatac-software/recipes-devtools/gitlab-runner/gitlab-runner_16.2.2.bb:do_compile) failed with exit code '1'
NOTE: Tasks Summary: Attempted 361 tasks of which 360 didn't need to be rerun and 1 failed.
```

The `go.bbclass` tries to pass all these lines as arguments to `go install` (as far as I can tell).

Reply:
I was wrong here. All of the verbose `HASH[…]` lines were output that `go list` outputs on stderr, so these did not get piped anywhere.

The underlying problem was that the `go.bbclass` does by default compile with the `-linkshared` flag, which for the gitlab-runner fails due to duplicate symbols in the generated binary.
When compiling without `-linkshared` the duplicate symbols do not seem to be an issue.
 
By setting `GO_LINKSHARED = ""` the `-linkshared` flag is omitted and the recipe builds without the ugly hack.